apis/v1/gatewayclass_types.go

```diff
@@ -93,8 +92,10 @@ type GatewayClassSpec struct {
 	// or an implementation-specific custom resource. The resource can be
 	// cluster-scoped or namespace-scoped.
 	//
-	// If the referent cannot be found, the GatewayClass's "InvalidParameters"
-	// status condition will be true.
+	// If the referent cannot be found, refers to an unsupported kind, or when
+	// the data within that resource is malformed, the GatewayClass will be
```

Comment:
```suggestion
	// the data within that resource is malformed, the GatewayClass may be
```
I don't think we enforce this behavior with any conformance tests currently, and I'm curious what some implementations that don't follow the "snapshotting" behavior do here - flipping a GatewayClass to `{ type: Accepted, status: False }` by setting a bad parameter could be pretty destructive to any Gateways with a "live" reference.

Reply:
Good catch - in general we try to use RFC language in the spec. I'd recommend using "SHOULD" in this context.

Reply:
```suggestion
	// the data within that resource is malformed, the GatewayClass SHOULD be
```